File: python/robobites_shared/money.py
from decimal import Decimal, ROUND_HALF_UP


def format_money(cents: int, currency: str = "USD") -> str:
    amount = Decimal(cents) / Decimal(100)
    return f"{currency} {amount.quantize(Decimal('0.01'))}"


def parse_money(value: str) -> int:
    normalized = "".join(ch for ch in value if ch.isdigit() or ch in ".-")
    amount = Decimal(normalized).quantize(Decimal("0.01"), rounding=ROUND_HALF_UP)
    return int(amount * 100)

File: ts/src/money/money.ts
export type Currency = "USD" | "CAD";

const formatters = new Map<string, Intl.NumberFormat>();

export function formatMoney(cents: number, currency: Currency = "USD", locale = "en-US"): string {
  const key = `${locale}:${currency}`;
  let formatter = formatters.get(key);
  if (!formatter) {
    formatter = new Intl.NumberFormat(locale, { style: "currency", currency });
    formatters.set(key, formatter);
  }
  return formatter.format(cents / 100);
}

export function parseMoney(input: string): number {
  const normalized = input.replace(/[^0-9.-]/g, "");
  const amount = Number.parseFloat(normalized);
  if (!Number.isFinite(amount)) {
    throw new Error("Money value is not numeric");
  }
  return Math.round(amount * 100);
}
